src/Parser.php

```diff
@@ -158,14 +168,70 @@ private function findLibrary(): string
             '/usr/lib/',
         ];
 
+        // First, try to find platform-specific library
         foreach ($possiblePaths as $path) {
             $filePath = $path . $libraryName;
             if (file_exists($filePath)) {
                 return $filePath;
             }
         }
 
-        throw new ParserException('libpubliccode-parser.so not found');
+        // Fall back to legacy library name (for backward compatibility)
+        foreach ($possiblePaths as $path) {
+            $filePath = $path . $legacyLibraryName;
+            if (file_exists($filePath)) {
+                return $filePath;
+            }
+        }
+
+        throw new ParserException(
+            "libpubliccode-parser library not found for platform: {$os}-{$arch}. " .
+            "Searched for: {$libraryName}"
+        );
+    }
+
+    /**
+     * Detect operating system
+     *
+     * @return string 'linux' or 'darwin'
+     * @throws ParserException
+     */
+    private function detectOS(): string
+    {
+        $osFamily = PHP_OS_FAMILY;
+
+        if ($osFamily === 'Linux') {
+            return 'linux';
+        }
+
+        if ($osFamily === 'Darwin') {
+            return 'darwin';
+        }
+
+        throw new ParserException("Unsupported operating system: {$osFamily}");
+    }
+
+    /**
+     * Detect CPU architecture
+     *
+     * @return string 'amd64' or 'arm64'
+     * @throws ParserException
+     */
+    private function detectArchitecture(): string
+    {
+        $machine = php_uname('m');
```

Comment:
Not super familiar with PHP, sorry. Does `php_uname` requires some additional extension or is it always available?